ariadne_codegen/contrib/no_global_imports.py

```diff
@@ -0,0 +1,359 @@
+                continue
+
+            if node.module is None:
+                continue
+
+            # We only care about local imports from our generated code.
+            if node.level != 1 and not node.module.startswith("."):
+                continue
+
+            for name in node.names:
+                from_ = "." * node.level + node.module
+                if isinstance(name, ast.alias):
+                    self.imported_classes[name.name] = from_
+
+    def _rewrite_input_args_to_constants(
+        self, method_def: ast.FunctionDef | ast.AsyncFunctionDef
+    ) -> ast.FunctionDef | ast.AsyncFunctionDef:
+        """Rewrite the arguments to a method.
+
+        For any `ast.Name` that requires an import convert it to an
+        `ast.Constant` instead. The actual class will be noted and imported
+        in a `TYPE_CHECKING` context.
+
+        :param method_def: Method definition
+        :returns: The same definition but updated
+        """
+        if not isinstance(method_def, (ast.FunctionDef, ast.AsyncFunctionDef)):
+            return method_def
+
+        for i, input_arg in enumerate(method_def.args.args):
+            annotation = input_arg.annotation
+            if isinstance(annotation, (ast.Name, ast.Subscript, ast.Tuple)):
+                method_def.args.args[i].annotation = self._update_name_to_constant(
+                    annotation
+                )
+
+        return method_def
+
+    def _insert_import_statement_in_method(
+        self, method_def: ast.FunctionDef | ast.AsyncFunctionDef
+    ):
+        """Insert import statement in method.
+
+        Each method will eventually pass the returned value to a class we've
+        generated. Since we only need it in the scope of the method ensure we
+        add it at the top of the method only. It will be removed from the global
+        scope.
+
+        :param method_def: The method definition to updated
+        """
+        # Find the last statement in the body, the call to this class is
+        # what we need to import first.
+        return_stmt = method_def.body[-1]
+        if isinstance(return_stmt, ast.Return):
+            call = self._get_call_arg_from_return(return_stmt)
+        elif isinstance(return_stmt, ast.AsyncFor):
+            call = self._get_call_arg_from_async_for(return_stmt)
+        else:
+            return
+
+        if call is None:
+            return
+
+        import_class = self._get_class_from_call(call)
+        if import_class is None:
+            return
+
+        import_class_id = import_class.id
+
+        # We add the class to our set of imported in methods - these classes
+        # don't need to be imported at all in the global scope.
+        self.imported_in_method.add(import_class.id)
+        method_def.body.insert(
+            0,
+            ast.ImportFrom(
+                module=self.imported_classes[import_class_id],
+                names=[import_class],
+            ),
```

Comment:
I developed this plugin outside of `ariadne-codegen` so I didn't use any helpers to generate these. Tbh I don't think it helps, it just makes it tricker to read.

Let me know if you'd rather me to import and use all the generators even for plugins or if I can keep this as is.